### PR TITLE
Introduce Endpoint and new StreamEntryID

### PR DESCRIPTION
**Pull request by sazzad16**
Extended version of #2540 

Resolves https://github.com/redis/jedis/projects/1#card-55852671 
Resolves https://github.com/redis/jedis/projects/1#card-55852669

### CODE REVIEW

File: src/main/java/redis/clients/jedis/params/XPendingParams.java [hunk 1]
```diff
@@ -30,16 +38,34 @@ public XPendingParams idle(long idle) {
     return this;
   }
 
+  /**
+   * @deprecated Use {@link #start(redis.clients.jedis.args.RangeEndpoint)}.
+   */
+  @Deprecated
   public XPendingParams start(StreamEntryID start) {
     this.start = start;
```
Comment (gkorland) [inline src/main/java/redis/clients/jedis/params/XPendingParams.java:46]:
```suggestion
    this.startEndpoint = RangeEndpoint.of(start);
```

Why do you need to deprecate it? it feels like in most cases it's easier to use it

File: src/main/java/redis/clients/jedis/params/XPendingParams.java [hunk 2]
```diff
@@ -52,6 +78,13 @@ public XPendingParams consumer(String consumer) {
 
   @Override
   public byte[][] getByteParams() {
+    if (start != null && startEndpoint == null) {
+      startEndpoint = RangeEndpoint.of(RangeEndpoint.convert(start));
```
Comment (gkorland) [inline src/main/java/redis/clients/jedis/params/XPendingParams.java:82]: If anyway you're going to convert `start` to `startEndpoint` why not do it in the setter?

File: src/main/java/redis/clients/jedis/args/StreamEntryID.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package redis.clients.jedis.args;
```
Comment (gkorland) [inline src/main/java/redis/clients/jedis/args/StreamEntryID.java:1]: @sazzad16 what about the original StreamEntryID class?

Comment (sazzad16) [reply src/main/java/redis/clients/jedis/args/StreamEntryID.java:1]: @gkorland Let's discuss...

**Pull request closed without merging**